src/inspector/plugin.ts

```diff
@@ -113,37 +109,49 @@ const service: JupyterLabPlugin<IInspector> = {
  */
 const consolePlugin: JupyterLabPlugin<void> = {
   id: 'jupyter.extensions.console-inspector',
-  requires: [IConsoleTracker],
-  provides: IInspector,
+  requires: [IInspector, IConsoleTracker],
   autoStart: true,
-  activate: (app: JupyterLab, consoles: IConsoleTracker): void => {
+  activate: (app: JupyterLab, manager: IInspector, consoles: IConsoleTracker): void => {
+    // Maintain association of new consoles with their respective handlers.
+    const handlers = new AttachedProperty<Widget, InspectionHandler>({
```

Comment:
can use a simple `Map` directly and remove entries from the map when the widget is disposed.